daemon/cmd/daemon_main.go

```diff
@@ -1647,110 +1643,17 @@ func initKubeProxyReplacementOptions() {
-			}
-			l.Info("Using auto-derived devices for BPF node port")
-		}
-	}
-
-	if option.Config.EnableNodePort &&
-		option.Config.NodePortAcceleration != option.NodePortAccelerationDisabled {
-		if option.Config.Tunnel != option.TunnelDisabled {
-			log.Fatalf("Cannot use NodePort acceleration with tunneling. Either run cilium-agent with --%s=%s or --%s=%s",
-				option.NodePortAcceleration, option.NodePortAccelerationDisabled, option.TunnelName, option.TunnelDisabled)
-		}
-
-		if option.Config.XDPDevice != "undefined" &&
-			(option.Config.DirectRoutingDevice == "" ||
-				option.Config.XDPDevice != option.Config.DirectRoutingDevice) {
-			log.Fatalf("Cannot set NodePort acceleration device: mismatch between Prefilter device %s and NodePort device %s",
-				option.Config.XDPDevice, option.Config.DirectRoutingDevice)
-		}
-		option.Config.XDPDevice = option.Config.DirectRoutingDevice
-		if err := loader.SetXDPMode(option.Config.NodePortAcceleration); err != nil {
-			log.WithError(err).Fatal("Cannot set NodePort acceleration")
-		}
-	}
-	if option.Config.EnableNodePort {
-		for _, iface := range option.Config.Devices {
-			link, err := netlink.LinkByName(iface)
-			if err != nil {
-				log.WithError(err).Fatalf("Cannot retrieve %s link", iface)
-			}
-			if strings.ContainsAny(iface, "=;") {
-				// Because we pass IPV{4,6}_NODEPORT addresses to bpf/init.sh
-				// in a form "$IFACE_NAME1=$IPV{4,6}_ADDR1;$IFACE_NAME2=...",
-				// we need to restrict the iface names. Otherwise, bpf/init.sh
-				// won't properly parse the mappings.
-				log.Fatalf("%s link name contains '=' or ';' character which is not allowed",
-					iface)
-			}
-			if idx := link.Attrs().Index; idx > math.MaxUint16 {
-				log.Fatalf("%s link ifindex %d exceeds max(uint16)", iface, idx)
-			}
-		}
-
-		if option.Config.EnableIPv4 &&
-			option.Config.Tunnel == option.TunnelDisabled &&
-			option.Config.NodePortMode != option.NodePortModeSNAT &&
-			len(option.Config.Devices) > 1 {
-
-			// In the case of the multi-dev NodePort DSR, if a request from an
-			// external client was sent to a device which is not used for direct
-			// routing, such request might be dropped by the destination node
-			// if the destination node's direct routing device's rp_filter = 1
-			// and the client IP is reachable via other device than the direct
-			// routing one.
-
-			iface := option.Config.Devices[0] // direct routing interface
-			if val, err := sysctl.Read(fmt.Sprintf("net.ipv4.conf.%s.rp_filter", iface)); err != nil {
-				log.Warnf("Unable to read net.ipv4.conf.%s.rp_filter: %s. Ignoring the check",
-					iface, err)
-			} else {
-				if val == "1" {
-					log.Warnf(`DSR might not work for requests sent to other than %s device. `+
-						`Run 'sysctl -w net.ipv4.conf.%s.rp_filter=2' (or set to '0') on each node to fix`,
-						iface, iface)
-				}
+				disableNodePort()
+				log.Warn(fmt.Sprintf("%s. Disabling BPF NodePort.", err))
```

Comment:
Consider using `log.WithError(err)` here?

```suggestion
				log.WithError(err).Warn("Disabling BPF NodePort.")
```